nest-tax-backend/src/card-payment-reporting/card-payment-reporting.service.ts

```diff
@@ -327,10 +328,17 @@ export class CardPaymentReportingService {
         if (!fileDate) return null
 
         const dateInfo = this.getDateInfo(fileDate)
+        if (dateInfo.today.isBefore(dayjs().subtract(180, 'day'))) {
+          return null
+        }
 
         // Parse CSV and add columns
         const csvData = this.processCsvData(file.content)
 
+        if (csvData.length >= 2 && csvData[1].transactionType === 'POHL') {
+          return null
+        }
```

Comment:
Many even self explanatory code block have a comment, but this advanced non-intuitive logic doesn't. Please, explain what is going on and why we do this. Also perhaps make `POHL` a constant with some sensible name to increase the code readability (which can make the comment shorter).